src/pubsub/ua_pubsub.c

```diff
@@ -5,6 +5,10 @@
  * Copyright (c) 2017-2018 Fraunhofer IOSB (Author: Andreas Ebner)
  */
 
+#include <ua_server_pubsub.h>
+#include "server/ua_server_internal.h"
+#include "ua_log_stdout.h"
```

Comment:
The same issue is also included in the ua pubsubmanager.c. Should we remove the dependency and include the server pointer as parameter to any PubSub method or what is your suggested solution?

Reply:
The same issue is also included in the ua pubsubmanager.c. Should we remove the dependency and include the server pointer as parameter to any PubSub method or what is your suggested solution?

Reply:
There is only one log instance in pubsub.c that can also be handled in the calling method since a status code is returned.
So the include is not needed at all in pubsub.c

In pubsub_manager.c, you always have a server pointer already.
Just go server->config->logger for the UA_Log call.